src/function/cast/string_cast.cpp

```diff
@@ -166,9 +167,73 @@ bool StringListCastLoop(string_t *source_data, ValidityMask &source_mask, Vector
 	       all_converted;
 }
 
-bool StringListCast(Vector &source, Vector &result, idx_t count, CastParameters &parameters) {
+static LogicalType InitVarcharStructType(const LogicalType &target) {
+	return LogicalType::STRUCT(
+	    child_list_t<LogicalType>(StructType::GetChildCount(target), make_pair("", LogicalType::VARCHAR)));
+}
+
+bool VectorStringToStruct::StringToNestedTypeCastLoop(string_t *source_data, ValidityMask &source_mask, Vector &result,
+                                                      ValidityMask &result_mask, idx_t count,
+                                                      CastParameters &parameters, const SelectionVector *sel) {
+
+	auto varchar_struct_type = InitVarcharStructType(result.GetType());
+	Vector varchar_vector(varchar_struct_type, count);
+	auto &child_vectors = StructVector::GetEntries(varchar_vector);
+	auto &result_children = StructVector::GetEntries(result);
+
+	string_map_t<idx_t> child_names;
+	vector<ValidityMask *> child_masks;
+	// to cover missing values, set all child masks to invalid, then only set the found values to valid
+	for (idx_t child_idx = 0; child_idx < result_children.size(); child_idx++) {
+		child_names.insert({StructType::GetChildName(result.GetType(), child_idx), child_idx});
+		if (result.GetVectorType() == VectorType::CONSTANT_VECTOR) {
+			(child_vectors[child_idx])->SetVectorType(VectorType::CONSTANT_VECTOR);
+			child_masks.emplace_back(&ConstantVector::Validity(*child_vectors[child_idx]));
+		} else {
+			child_masks.emplace_back(&FlatVector::Validity(*child_vectors[child_idx]));
+		}
+		child_masks[child_idx]->SetAllInvalid(count);
+	}
+
+	bool all_converted = true;
+	for (idx_t i = 0; i < count; i++) {
+		idx_t idx = i;
+		if (sel) {
+			idx = sel->get_index(i);
+		}
+		if (!source_mask.RowIsValid(idx)) {
+			result_mask.SetInvalid(i);
+			continue;
+		}
+		if (!VectorStringToStruct::SplitStruct(source_data[idx], child_vectors, i, child_names, child_masks)) {
+			string text = "Type VARCHAR with value '" + source_data[idx].GetString() +
+			              "' can't be cast to the destination type STRUCT";
+			for (auto &child_mask : child_masks) {
+				child_mask->SetInvalid(idx); // some values may have already been found and set valid
+			}
+			HandleVectorCastError::Operation<string_t>(text, result_mask, idx, parameters.error_message, all_converted);
+		}
+	}
+
+	auto &cast_data = (StructBoundCastData &)*parameters.cast_data;
+
+	for (idx_t child_idx = 0; child_idx < result_children.size(); child_idx++) {
+		auto &varchar_vector = *child_vectors[child_idx];
+		auto &result_child_vector = *result_children[child_idx];
+		CastParameters child_parameters(parameters, cast_data.child_cast_info[child_idx].cast_data.get());
```

Comment:
(very nitpicky, sorry)
can we add a reference variable to abbreviate this big line?
i.e
```cpp
auto& child_cast_info = cast_data.child_cast_info[child_idx];
```
It can also be used below then

Reply:
will do!